src/Nancy/Json/Simple/NancySerializationStrategy.cs

```diff
@@ -0,0 +1,231 @@
+            var genericArguments = type.GetGenericArguments();
+
+            for (var i = 0; i < genericArguments.Length; i++)
+            {
+                var deserializedObject = this.DeserializeObject(valueDictionary.Values.ElementAt(i),
+                    genericArguments[i]);
+
+                values.Add(valueDictionary.Keys.ElementAt(i), deserializedObject);
+            }
+
+            return genericTypeConverter.Deserialize(values, type);
+        }
+
+        /// <summary>
+        /// Serialize an object
+        /// </summary>
+        /// <param name="input">The object to serialize</param>
+        /// <param name="output">The serialized object</param>
+        /// <returns></returns>
+        protected override bool TrySerializeKnownTypes(object input, out object output)
+        {
+            var dynamicValue = input as DynamicDictionaryValue;
+            if (!ReferenceEquals(dynamicValue, null) && dynamicValue.HasValue)
+            {
+                output = dynamicValue.Value;
+                return true;
+            }
+
+            var inputType = input.GetType().GetTypeInfo();
+            if (this.TrySerializeJavaScriptConverter(input, out output, inputType))
+            {
+                return true;
+            }
+
+            if (this.TrySerializePrimitiveConverter(input, ref output, inputType))
+            {
+                return true;
+            }
+
+            var type = input as Type;
+            if (type != null)
+            {
+                output = type.GetTypeInfo().FullName;
+                return true;
+            }
+
+            if (input is DateTime)
+            {
+                return this.SerializeDateTime((DateTime)input, out output);
+            }
+
+            if (input is DateTimeOffset)
+            {
+                var dto = (DateTimeOffset)input;
+                output = dto.ToString("o", CultureInfo.InvariantCulture);
+                return true;
+            }
+
+            return base.TrySerializeKnownTypes(input, out output);
+        }
+
+        private bool SerializeDateTime(DateTime input, out object output)
+        {
+            var dateTime = input;
+            if (dateTime.Kind == DateTimeKind.Unspecified)
+            {
+                dateTime = new DateTime(dateTime.Ticks, DateTimeKind.Local);
+            }
+
+            output = dateTime.ToString("o", CultureInfo.InvariantCulture);
+
+            return true;
+        }
+
+        private bool TrySerializePrimitiveConverter(object input, ref object output, Type inputType)
+        {
+            var primitiveConverter = this.FindPrimitiveConverter(inputType);
+            if (primitiveConverter == null)
+            {
+                return false;
+            }
+
+            output = primitiveConverter.Serialize(input);
+            return true;
+        }
+
+        private JavaScriptPrimitiveConverter FindPrimitiveConverter(Type inputType)

```

Comment:
What do you guys think about adding caching on this method and `FindJavaScriptConverter`? That would avoid us doing the same kind of LINQ query each a Type is serialized. 

We could just use `ConcurrentDictionary<Type, JavaScriptPrimitiveConverter>` and use the `GetOrAdd` pattern?